app/modules/hmc5883l.c

```diff
@@ -106,11 +176,22 @@ static int hmc5883_read(lua_State* L) {
     return 3;
 }
 
+static int xmc5883_read(lua_State* L) {
+    uint8_t i2c_addr;
+    i2c_addr = xmc5883_identify();
```

Comment:
This introduces a delay on each call of read(). Why not save the i2c_addr in xmc5883_i2c_addr (as part of setup/init) and then just access the right address immediately.

Reply:
I was considering approach you suggested, but I wasn't really sure if nodemcu modules stores their states between each LUA command call - that's why I decided to determine chip type each time. But if it's possible to store such value, then I'll do it the way you suggest.